diesel/src/connection/statement_cache.rs

```diff
@@ -0,0 +1,54 @@
+use std::any::TypeId;
+use std::borrow::Cow;
+
+use backend::Backend;
+use query_builder::*;
+use result::QueryResult;
+
+#[doc(hidden)]
+#[allow(missing_debug_implementations)]
+#[derive(Hash, PartialEq, Eq)]
+pub enum StatementCacheKey<DB: Backend> {
+    Type(TypeId),
+    Sql {
+        sql: String,
+        bind_types: Vec<DB::TypeMetadata>,
+    }
+}
```

Comment:
`TypeId` is as large as one `u64`, while `Sql` is 6 `usize`s. I'd be surprised if it matters perf-wise even in large apps but you mentioned `Type` should be the hot path anyway.